core/modules.rs

```diff
@@ -468,11 +464,23 @@ impl Modules {
     self.by_name.is_alias(name)
   }
 
-  pub fn get_info(&self, id: ModuleId) -> Option<&ModuleInfo> {
-    if id == 0 {
-      return None;
+  pub fn get_handle(&self, id: ModuleId) -> Option<v8::Global<v8::Module>> {
+    self.handles.get(&id).cloned()
+  }
+
+  pub fn get_info(
+    &self,
+    global: &v8::Global<v8::Module>,
+  ) -> Option<&ModuleInfo> {
+    // NOTE(@bartlomieju): not ideal, but I was unable to create
+    // another hashmap indexed by `handle`.
```

Comment:
If (and only if) you remember: what was the error you got? `Global` implements `PartialEq` and `Hasher` so it should be possible to use them as keys.

Reply:
I thought so too, here's the error: https://github.com/denoland/deno/pull/8354#discussion_r522157813

Reply:
That's because v8__Data__GetHash doesn't work right for Module handles.
cc @bnoordhuis

Reply:
@piscisaureus I think it was already fixed in https://github.com/denoland/rusty_v8/pull/522 just waiting on the release